src/AutoUI/index.tsx

```diff
@@ -379,7 +379,7 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 		});
 	};
 
-	if (!!loading) {
+	if (!!loading && (!data || (Array.isArray(data) && !data?.length))) {
```

Comment:
Shouldn't this be v ?
```suggestion
	if (loading && data == null) {
```
I mean, `Array.isArray(data) && !data.length` is a totally valid case, which we do need afaik so that we show the "you have no devices" message instead of the spinner. Or do we accidentally/incorrectly coalesce the `null` to an empty array somewhere?